Src/Core/Collections/CollectionAssertions.cs

```diff
@@ -209,65 +207,33 @@ public AndConstraint<TAssertions> Equal(IEnumerable expected, string because = "
         protected void AssertSubjectEquality<TActual, TExpected>(IEnumerable expectation, Func<TActual, TExpected, bool> predicate,
             string because = "", params object[] reasonArgs)
         {
-            AssertionScope assertion = Execute.Assertion.BecauseOf(because, reasonArgs);
-
-            bool subjectIsNull = ReferenceEquals(Subject, null);
-            bool expectationIsNull = ReferenceEquals(expectation, null);
 
+            var subjectIsNull = ReferenceEquals(Subject, null);
```

Comment:
Please avoid using `var`, unless the type is explicitly mentioned. Same for all other changes in this PR.